### PR TITLE
Updates to CA test processes

### PR DESCRIPTION
**Pull request by clintwilson**
This commit updates the openssl.cnf file with new sections for each of
the different types of certs in the WInnForum hierarchy. The
configuration uses the version of the ComSec document approved/published
by WG2.
This version of the configuration includes the WInnForum registered OIDs
for ROLE as identified/defined in the ComSec document.
There is also an addition of a PowerShell script. This script should be
able to be run on a Windows machine that has OpenSSL installed. It will
first prompt for your OpenSSL /bin directory path and the name of your
OpenSSL config filename. From there, it should create in your /bin
folder 11 private keys, CSRs, and signed certificates. There are
numerous variables in the .ps1 file itself which can be modified to
update what types of certs are created.
Last, this commit also adds a folder labeled "ExamplePKI" which contains
the certificate files which result from running the PowerShell script.

### CODE REVIEW

File: cert/openssl.cnf [hunk 1]
```diff
@@ -1,234 +1,424 @@
...
-sas_fccid_oid = 1.3.6.1.4.1.46609.1.2
-sas_serialnum_oid = 1.3.6.1.4.1.46609.1.3
-sas_maxgrants_oid = 1.3.6.1.4.1.46609.1.4
-sas_phase_oid = 1.3.6.1.4.1.46609.1.5
-sas_zone_oid = 1.3.6.1.4.1.46609.1.6
-sas_freq_oid = 1.3.6.1.4.1.46609.1.7
-sas_test_oid = 1.3.6.1.4.1.46609.1.999
+#sas_phase_oid = 1.3.6.1.4.1.46609.1.5
+#sas_zone_oid = 1.3.6.1.4.1.46609.1.6
+#sas_freq_oid = 1.3.6.1.4.1.46609.1.7
```
Comment (gregbillock) [inline cert/openssl.cnf]: Do you want to just delete these lines?

File: cert/openssl.cnf.old [hunk 2]
```diff
@@ -0,0 +1,234 @@
+# SAS OpenSSL configuration file.
```
Comment (gregbillock) [inline cert/openssl.cnf.old]: No need to keep this file -- you can just delete it.

Are there other files in the sasCA directory that should be deleted as well?

File: cert/PowerShell_CACreationScript.ps1 [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+#Device Manufacturer Intermediate
+$dm = "wif_dm_ca"
+$dm_sign = $dm + $sig
+
+Invoke-Expression "openssl req -new -newkey rsa:$intRSAKeysize -nodes -reqexts $dm -out $dm.csr -keyout $dm.pkey -subj `"/C=US/ST=District of Columbia/L=Washington/O=Wireless Innovation Forum/OU=www.wirelessinnovation.org/CN=WInnForum RSA Device Manufacturer CA-1`" -config `"$openssl\$cnf`""
+Invoke-Expression "openssl ca -create_serial -cert $root.crt -keyfile $root.pkey -outdir `"./root/ca/newcerts`" -batch -out $dm.crt -utf8 -days $intValidity -policy policy_anything -md sha384 -extensions $dm_sign -config `"$openssl\$cnf`" -in $dm.csr"
+
+#Device Manufacturer EE
+$dm_ee = "wif_dm_req"
```
Comment (gregbillock) [inline cert/PowerShell_CACreationScript.ps1]: Is this the intermediate for the device manufacturer or the CBSD (device) EE cert? If it's the EE cert, let's label it 'wif_cbsd_req'

File: cert/openssl.cnf [hunk 4]
```diff
@@ -1,234 +1,424 @@
...
+#FREQUENCY = 
+#STARTDATETIME = 
 
-[ v3_ca ]
-# This section is used when self-signing a (test) CA cert
-# PKIX recommendation.
-subjectKeyIdentifier=hash
-authorityKeyIdentifier=keyid:always,issuer:always
+####################################################################
+[ wif_dm_req ]
```
Comment (gregbillock) [inline cert/openssl.cnf]: I think this is wif_cbsd_req as above

File: cert/openssl.cnf [hunk 5]
```diff
@@ -1,234 +1,424 @@
...
+[ ocsp_section ] 
 
-emailAddress			= Email Address
-emailAddress_max		= 64
+caIssuers;URI.0 = http://cacerts.digicert.com/WinnForumCA.crt
+OCSP;URI.0 = http://ocsp.digicert.com
 
-[ req_attributes ]
+####################################################################
+[ wif_root_ca ]
```
Comment (gregbillock) [inline cert/openssl.cnf]: These are really great. The different sections are a great way to spell this out.

**Comment by gregbillock** [comment]
This looks great. Thanks. I had just a couple in-line comments on the wif_dm_req terminology to distinguish a device from a manufacturer intermediate.

**Comment by gregbillock** [comment]
Looks good! Thanks.

**Comment by clintwilson** [comment]
Awesome feedback. Made the suggested changes in a second commit.

**Comment by gregbillock** [comment]
Cool. Go ahead and merge when you're ready.

File: cert/PowerShell_CACreationScript.ps1 [hunk 6]
```diff
@@ -0,0 +1,105 @@
+﻿#Get to OpenSSL directory
```
Comment (gregbillock) [inline cert/PowerShell_CACreationScript.ps1]: Since this is a script, can you add the repo preamble (see CONTRIBUTING in the root dir)

**Comment by gregbillock** [comment]
Actually wait a sec. It looks like there's a way to comment ps1 files; can you add the repo file header there?

**Pull request merged**